include/osmium/thread/pool.hpp

```diff
@@ -162,7 +162,11 @@ namespace osmium {
 
             static constexpr int default_num_threads = 0;
 
-            static Pool& instance() {
+            explicit Pool(int num_threads = default_num_threads) :
```

Comment:
Good point! How about this: make the default value `0`. Check inside the constructor if it is and then use the `detail::get_work_queue_size()`. The the documentation should say: "Set to 0 to get max queue size from OSMIUM_MAX_WORK_QUEUE_SIZE env variable". The the detail is hidden behind a good default value with the added benefit that we'll never have an actual queue size of 0 which would make no sense.